db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2210 @@
+
+    expectedResults = List.of(
+        new DerivedEvent(
+            "1",
+            "DerivationA",
+            "Derivation_Test_01.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-01 00:00:00+00",
+            "02:10:00",
+            "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 12:00:00+00\"),[\"2024-01-02 12:00:00+00\",\"2024-01-03 00:00:00+00\")}",
+            "2024-01-19 00:00:00+00"
+        ),
+        new DerivedEvent(
+            "9",
+            "DerivationC",
+            "Derivation_Test_03.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-02 00:00:00+00",
+            "01:00:00",
+            "{[\"2024-01-01 12:00:00+00\",\"2024-01-02 12:00:00+00\")}",
+            "2024-01-21 00:00:00+00"
+        ),
+        new DerivedEvent(
+            "3",
+            "DerivationB",
+            "Derivation_Test_01.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-02 23:00:00+00",
+            "03:00:00",
+            "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 12:00:00+00\"),[\"2024-01-02 12:00:00+00\",\"2024-01-03 00:00:00+00\")}",
+            "2024-01-19 00:00:00+00"
+        ),
+        new DerivedEvent(
+            "5",
+            "DerivationC",
+            "Derivation_Test_02.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-05 23:00:00+00",
+            "01:10:00",
+            "{[\"2024-01-03 00:00:00+00\",\"2024-01-10 00:00:00+00\")}",
+            "2024-01-20 00:00:00+00"
+        ),
+        new DerivedEvent(
+            "6",
+            "DerivationC",
+            "Derivation_Test_02.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-06 12:00:00+00",
+            "02:00:00",
+            "{[\"2024-01-03 00:00:00+00\",\"2024-01-10 00:00:00+00\")}",
+            "2024-01-20 00:00:00+00"
+        ),
+        new DerivedEvent(
+            "2",
+            "DerivationB",
+            "Derivation_Test_02.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-09 11:00:00+00",
+            "01:05:00",
+            "{[\"2024-01-03 00:00:00+00\",\"2024-01-10 00:00:00+00\")}",
+            "2024-01-20 00:00:00+00"
+        ),
+        new DerivedEvent(
+            "8",
+            "DerivationB",
+            "Derivation_Test_00.json",
+            DERIVATION_GROUP + "_2",
+            "2024-01-10 11:00:00+00",
+            "01:05:00",
+            "{[\"2024-01-10 00:00:00+00\",\"2024-01-11 00:00:00+00\")}",
+            "2024-01-18 00:00:00+00"
+        )
+    );
+    assertEquals(expectedResults.size(), results.size());
+    assertTrue(results.containsAll(expectedResults));
+  }
+
+  /**
+   * This test ensures that a derivation group can be associated with a plan. This behavior is very simple and doesn't
+   *    have many derivative cases besides merging of plans, and that behavior is handled in a test in
+   *    PlanCollaborationTests.java.
+   */
+  @Test
+  void associateDerivationGroupWithBasePlan() throws SQLException {
```

Comment:
This test is not meaningful. Meaningful tests involving the `pdg` table would be as follows:
 - `last_acknowledged_at` updates correctly when `acknowledged` is set to `true`
 - `last_acknowledged_at` does not update when `acknowledged` is set to `false`
 - `acknowledged` is correctly set to `false` when a new source is added to a derivation group (and `acknowledged` was `true` prior to the insert).

Reply:
Committed!